hil-test/tests/init.rs

```diff
@@ -0,0 +1,89 @@
+//! Initialization tests
+
+//% CHIPS: esp32 esp32c2 esp32c3 esp32c6 esp32h2 esp32s2 esp32s3
+
+#![no_std]
+#![no_main]
+
+use esp_hal::{
+    config::WatchdogStatus,
+    delay::Delay,
+    prelude::*,
+    rtc_cntl::Rtc,
+    timer::timg::TimerGroup,
+    Config,
+};
+use hil_test as _;
+
+#[cfg(test)]
+#[embedded_test::tests]
+mod tests {
+    use super::*;
+
+    #[test]
+    #[timeout(3)]
+    fn test_feeding_timg0_wdt() {
+        let peripherals = esp_hal::init({
+            let mut config = Config::default();
+            config.watchdog.timg0 =
+                WatchdogStatus::Enabled(fugit::MicrosDurationU64::millis(500 as u64));
+            config
+        });
+
+        let timg0 = TimerGroup::new(peripherals.TIMG0);
```

Comment:
Can we add tests for TIMG1 too, as that's the one I'm actually most worried won't work (because we actually reset  the peripheral on enable)